app/src/main/java/com/mutualmobile/praxis/ui/screen/home/HomeScreen.kt

```diff
@@ -0,0 +1,95 @@
+package com.mutualmobile.praxis.ui.screen.home
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.ExperimentalAnimationApi
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.mutualmobile.praxis.R
+import com.mutualmobile.praxis.domain.model.Joke
+import com.mutualmobile.praxis.ui.component.TopBar
+
+/**
+ * Created by Rooparsh Kalia on 31/07/21
+ */
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+fun HomeScreen(modifier: Modifier = Modifier,
+               viewModel: HomeVM,
+               onShowJokeClick: (List<Joke>) -> Unit,
+               onAboutClick: () -> Unit) {
+
+    val state by remember { viewModel.viewState }
+
+    if (state is HomeViewState.ShowJokes) {
+        onShowJokeClick((state as HomeViewState.ShowJokes).jokes)
+    }
+
+    HomeScreen(modifier = modifier,
+            state = state,
+            onShowJokeClick = { viewModel.loadJokes() },
+            onAboutClick = { onAboutClick() })
+}
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+@Preview("Initial State")
+fun HomeScreenInitialPreview() {
+    HomeScreen(state = HomeViewState.Initial,
+            onShowJokeClick = { },
+            onAboutClick = { }
+    )
+}
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+@Preview("Loading State")
+fun HomeScreenLoadingPreview() {
+    HomeScreen(state = HomeViewState.Loading,
+            onShowJokeClick = { },
+            onAboutClick = { }
+    )
+}
+
+@ExperimentalAnimationApi
+@Composable
+fun HomeScreen(modifier: Modifier = Modifier,
+               state: HomeViewState,
+               onShowJokeClick: () -> Unit,
+               onAboutClick: () -> Unit) {
+    Scaffold(topBar = { TopBar() }) {
```

Comment:
@rooparsh Should we put this Scaffold to the MainActivity somehow like this: 
```

BaseComposeTheme {
   Scaffold(
          appBar: xyz,
          bottomBar: xyz
      ) {
            NavGraph()
      }
       
 }
```
So that the scaffold can become Global. This can help when we have a bottom navigation kind of initial screen which have more than one bottom tabs.
You can see the implementation in JetSnack(https://github.com/android/compose-samples/tree/main/Jetsnack )
cc: @anmolvermamm